### PR TITLE
Setup parametric module with plotting

### PR DESCRIPTION
**Pull request by earoy**
For new tractometry repo: Added a new module parametric.py that performs OLS and LME modeling over the nodes of a given tract to allow for group comparisons, while controlling for potential covariates. Additionally, added another plotting function to plot.py that allows the user to visualize the "tract profiles" generated by the linear models and updated the plot_als_comparison.py example to demonstrate how to use these functions.

### CODE REVIEW

**Comment by arokem** [comment]
Hey @earoy : thanks for moving the PR here! 

As you can see, we now have a particularly persnickety linting system, so the CI is reporting on several issues with the code (line lengths, etc.). 

To set up this linting to operate locally on your code, so you can fix this up locally, you can install a pre-commit hook, by issuing the following: 

```
pip install -e .'[all]'
```

followed by: 

```
pre-commit install
``` 

Thereafter, each time you run `git commit` the pre-commit hook will check the code that you are committing and will either automatically fix it (whenever it can) or report the linting errors for you to fix locally before committing and pushing. I hope that makes sense, but let me know if it does not. We'll get there! The good news is that the documentation example now runs cleanly, so that's progress.

**Comment by earoy** [comment]
Hi @arokem! I installed the pre-commit hook and made some minor changes to the comments so that I could make a new commit...Looks like things failed again, seemingly because some lines are too long, but I'm struggling to read the error message to figure out which lines are causing the issue and when I look at the code locally every line seems to be under 88 characters

**Comment by arokem** [comment]
That's annoying. Let me see if I can sort this out locally on my machine.

**Comment by arokem** [comment]
OK - I believe I have resolved the formatting issues, and I will take a look at the rest later today. If you want to make any additional changes here, you'll want to issue a `git pull origin setup-parametric-module-with-plotting` before moving on.

File: afqinsight/parametric.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        A dictionary with the following key-value pairs:
+
+        {'tract': tract,
+                  'reference_coefs': coefs_default,
+                  'group_coefs': coefs_treat,
+                  'reference_CI': cis_default,
+                  'group_CI': cis_treat,
+                  'pvals': pvals,
+                  'reject_idx': reject_idx,
+                  'model_fits': fits}
```
Comment (arokem) [inline afqinsight/parametric.py:62]:
```suggestion
                 {'tract': tract,
                  'reference_coefs': coefs_default,
                  'group_coefs': coefs_treat,
                  'reference_CI': cis_default,
                  'group_CI': cis_treat,
                  'pvals': pvals,
                  'reject_idx': reject_idx,
                  'model_fits': fits}
```

**Comment by arokem** [comment]
Actually, this all looks great to me, so I will go ahead and merge when the CI is done.

**Comment by earoy** [comment]
Sounds good. Thanks for your help with the formatting troubles!

**Pull request merged**